service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -1,17 +1,120 @@
 package uk.gov.hmcts.reform.fpl.model;
 
-import lombok.AllArgsConstructor;
+import lombok.AccessLevel;
 import lombok.Builder;
 import lombok.Data;
+import lombok.Getter;
+import uk.gov.hmcts.reform.fpl.enums.OrderType;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentSocialWorkOther;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasDocumentStatus;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasEPOGrounds;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasMainApplicant;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasThresholdDetails;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasThresholdReason;
 
 import java.util.List;
 
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
 @Data
-@Builder(toBuilder = true)
-@AllArgsConstructor
+@Builder
+@HasEPOGrounds
+@HasThresholdReason
+@HasThresholdDetails
+@HasDocumentStatus
+@HasMainApplicant
+@SuppressWarnings("membername")
 public class CaseData {
+    @NotBlank(message = "Enter a case name")
+    private final String caseName;
+    private final String gatekeeperEmail;
+    private final String caseLocalAuthority;
+    private final Risks risks;
+    @NotNull(message = "You need to add details to orders and directions needed")
+    @Valid
+    private final Orders orders;
+
+    public boolean hasEPOGrounds() {
```

Comment:
That method is used only in validator, IMO there is no reason to make add it to case data.

Reply:
It was to stop repeating the same logic in various validators. Is it better to repeat logic?

Reply:
I did not notice it. How many validators use it?

Reply:
Three in total
HasEpoGroundsValidator
HasThresholdDetailsValidator
HasThresholdReasonValidator

Reply:
Ok, I would still consider to remove it, there is nothing complex, just traversal through object structure. It could possibly be moved to base EPO validator.

Also name itself does not make sense. It says `hasEPOGrounds` while it does not check whether EPO grounds for application are populated. It basically checks whether EPO is selected among order types.

It would make bit more sense to expose null safe getter for order types (although still nasty).

Finally why didn't you resolve it by using validation groups? It would make sense to group EPO validation rules together and trigger alongside default ones when EPO is selected.

Reply:
Groups does not make sense and I explored but validation groups work in such a way that if one check fails, the others do not get run. I need all to run to show multiple validation messages.